src/components/diagrams/networkAreaDiagram/network-area-diagram.js

```diff
@@ -0,0 +1,442 @@
+                    updateLoadingState(false);
+                });
+        } else {
+            setSvg(noSvg);
+        }
+    }, [svgUrl, forceState, snackError]);
+
+    const updateNad = useCallback(() => {
+        if (svgRef.current) {
+            forceUpdate();
+        }
+    }, [forceUpdate]);
+
+    useEffect(() => {
+        if (studyUpdatedForce.eventData.headers) {
+            if (
+                studyUpdatedForce.eventData.headers['updateType'] ===
+                    'loadflow' ||
+                studyUpdatedForce.eventData.headers['updateType'] === 'study'
+            ) {
+                updateNad();
+            }
+        }
+        // Note: studyUuid, and loadNetwork don't change
+    }, [studyUpdatedForce, updateNad]);
+
+    useLayoutEffect(() => {
+        if (svg.svg) {
+            const nad = new NetworkAreaDiagramViewer(
+                svgRef.current,
+                svg.svg,
+                minWidth,
+                minHeight,
+                maxWidth,
+                maxHeight
+            );
+            setSvgPreferredHeight(nad.getHeight());
+            setSvgPreferredWidth(nad.getWidth());
+        }
+    }, [network, svg, workingNode, theme, nadId, ref, svgUrl]);
+
+    useLayoutEffect(() => {
+        if (
+            typeof svgFinalWidth != 'undefined' &&
+            typeof svgFinalHeight != 'undefined'
+        ) {
+            const divElt = svgRef.current;
+            if (divElt != null) {
+                const svgEl = divElt.getElementsByTagName('svg')[0];
+                if (svgEl != null) {
+                    svgEl.setAttribute(
+                        'width',
+                        fullScreen ? totalWidth - 40 : svgPreferredWidth
+                    );
+                    svgEl.setAttribute(
+                        'height',
+                        fullScreen ? totalHeight - 40 : svgPreferredHeight
+                    );
+                }
+            }
+        }
+    }, [
+        fullScreen,
+        svgFinalWidth,
+        svgFinalHeight,
+        svgPreferredWidth,
+        svgPreferredHeight,
+        totalWidth,
+        totalHeight,
+        svg,
+        theme,
+    ]);
+
+    const onCloseHandler = () => {
+        if (onClose !== null) {
+            dispatch(fullScreenNetworkAreaDiagram(undefined));
+            onClose(nadId);
+            setDepth(0);
+        }
+    };
+
+    const showFullScreen = () => {
+        dispatch(fullScreenNetworkAreaDiagram(nadId));
+    };
+
+    const hideFullScreen = () => {
+        dispatch(fullScreenNetworkAreaDiagram(undefined));
+    };
+
+    let sizeWidth, sizeHeight;
+    if (svg.error) {
+        sizeWidth = errorWidth; // height is not set so height is auto;
+    } else if (
+        typeof finalPaperWidth != 'undefined' &&
+        typeof finalPaperHeight != 'undefined'
+    ) {
+        sizeWidth = finalPaperWidth;
+        sizeHeight = finalPaperHeight;
+    }
+
+    return !svg.error ? (
```

Comment:
May be inverse the test to have the shortest branch first ?

Reply:
Done